File: AVS/task2/src/animals/bird/Bird.cpp
//
// Created by Матвей Попов on 14.10.2021.
//

#include "Bird.h"

Bird::Bird(const std::string &name, const int32_t &weight): Animal(name, weight) {}

Bird::Bird(const Bird &bird): Animal(bird) {
    isTransferable = new bool(bird.isTransferable);
}

Bird::~Bird() noexcept {
    delete isTransferable;
}

Bird& Bird::operator=(const Bird &bird) {
    Animal::operator=(bird);
    isTransferable = new bool (bird.isTransferable);
    return *this;
}

Animal* Animal::copy() {
    Bird* currentBird = dynamic_cast<Bird*>(this);
    Bird* copyBird = new Bird(*name, *weight);
    copyBird->setIsTransferable(currentBird->getIsTransferable());
    return copyBird;
}

bool Bird::getIsTransferable() const {
    return *isTransferable;
}

void Bird::setIsTransferable(const bool& isTransferable) {
    this->isTransferable = &isTransferable
}

File: AVS/task3/src/animals/bird/Bird.py
from src.animals.animal.Animal import Animal


class Bird(Animal):

    def __init__(self, name, weight, is_transferable):
        super(Bird, self).__init__(name, weight)
        self.is_transferable = is_transferable
        if is_transferable is not bool:
            print("Invalid is transferable parameter\n")
            print("Setting parameter to default: true\n")
            self.is_transferable = True

    def get_is_transferable(self):
        return self.is_transferable

    def get_info_data(self):
        return "This is is bird\n" + \
               "Is transferable " + str(self.is_transferable) + "\n" + \
               super(Bird, self).get_info_data()
